### PR TITLE
Feat: support Non-Admin API key creation

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Updated integration test, to test api-key team assignment.
Added API calls to add support for api-key project assignment.

Part of #405.

### CODE REVIEW

File: tests/integration/020_api_key/main.tf [hunk 1]
```diff
@@ -11,10 +11,20 @@ resource "env0_api_key" "test_api_key" {
...
+  name              = "my-little-user-api-key-${random_string.random.result}"
   organization_role = "User"
 }
 
+resource "env0_team" "team_resource" {
+  name        = "team-with-api-key"
+  description = "description"
+}
+
+resource "env0_user_team_assignment" "api_key_team_assignment" {
```
Comment (TomerHeber) [inline tests/integration/020_api_key/main.tf:23]: This is for testing team api-key assignment (Which is just user team assignment).

File: client/api_key_project_assignment.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package client
+
+type ApiKeyRole string
+
+const (
+	ApiKeyRoleDeployer ApiKeyRole = "Deployer"
+	ApiKeyRoleViewer   ApiKeyRole = "Viewer"
+	ApiKeyRolePlanner  ApiKeyRole = "Planner"
+	ApiKeyRoleAdmin    ApiKeyRole = "Admin"
+)
```
Comment (avnerenv0) [inline client/api_key_project_assignment.go:10]: we've got these also defined in `client/team_project_assignment.go`
Those are the same roles a regular user/team have. Can we reuse them?

File: client/api_key_project_assignment.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+type AssignAPIKeyToProjectPayload struct {
+	UserId string     `json:"userId"`
+	Role   ApiKeyRole `json:"role"`
+}
+
+type ApiKeyProjectAssignment struct {
+	Id string `json:"id"`
+}
+
+func (client *ApiClient) AssignApiKeyToProject(projectId string, payload *AssignAPIKeyToProjectPayload) (*ApiKeyProjectAssignment, error) {
```
Comment (avnerenv0) [inline client/api_key_project_assignment.go:21]: These endpoints aren't specific to api keys, they are for assigning users to projects, and a "non admin api key" is just a user at the end of it. 

WDYT about renaming this file/methods to reflect that?

File: tests/integration/020_api_key/main.tf (hunk 1 above)
Comment (avnerenv0) [reply tests/integration/020_api_key/main.tf:23]: not sure if it's part of the scope of this PR, but as part of #405 can we document somewhere that the `user_id` can also be an `api_key.id` ?

**Review by avnerenv0 — APPROVED** [review]
LGTM, comments for your consideration

File: client/api_key_project_assignment.go (hunk 3 above)
Comment (TomerHeber) [reply client/api_key_project_assignment.go:21]: I think it's a good idea. I'll make some changes.

File: client/api_key_project_assignment.go (hunk 2 above)
Comment (TomerHeber) [reply client/api_key_project_assignment.go:10]: Yes.

**Comment by avnerenv0** [comment]
Go for it
<div><br/>via <a href="https://giphy.com/thefastsaga/">The Fast Saga</a> on <a href="https://giphy.com/gifs/thefastsaga-fast-and-furious-dnCNI8LEf7K4otcMqS">GIPHY</a></div>

**Pull request merged**